casper/src/main/scala/coop/rchain/casper/blocks/proposer/Proposer.scala

```diff
@@ -128,10 +131,76 @@ object Proposer {
         maxSeqNum  = latestMsgs.find(_.sender == sender).map(_.senderSeq)
       } yield maxSeqNum.getOrElse(-1)
 
-    def createBlock(validatorIdentity: ValidatorIdentity) =
+    def createBlock(validatorIdentity: ValidatorIdentity): F[BlockCreatorResult] =
       for {
-        parentsPreState <- MultiParentCasper.getPreStateForNewBlock
-        result          <- BlockCreator.create(parentsPreState, validatorIdentity, shardId, dummyDeployOpt)
+        // merge pre state
+        preState <- MultiParentCasper.getPreStateForNewBlock
+        // misc
+        preStateHash      = preState.preStateHash
+        creatorsPk        = validatorIdentity.publicKey
+        creatorsId        = ByteString.copyFrom(creatorsPk.bytes)
+        creatorsLatestOpt = preState.justifications.find(_.sender == creatorsId)
+        nextSeqNum        = creatorsLatestOpt.map(_.seqNum + 1).getOrElse(0L)
+        nextBlockNum      = preState.justifications.map(_.blockNum).max + 1
+        parentHashes      = preState.justifications.map(_.blockHash)
+        finalBonds        = preState.fringeBondsMap
+        offenders         = preState.justifications.filter(_.validationFailed).map(_.sender)
```

Comment:
This data should be extracted and used in block validation. From parents state expected data should be created and be used as a basis for validation. Block metadata will store expected data and invalid parts like invalid block number will be store separately. This is to make DB consistent even with invalid blocks.